### PR TITLE
WFLY-16752 Make it possible for Hibernate ORM Dialects to be able to access org.postgresql.jdbc and com.oracle.ojdbc JDBC driver classes

### PR DESCRIPTION
**Pull request by scottmarlow**
https://issues.redhat.com/browse/WFLY-16752

### CODE REVIEW

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/hibernate/main/module.xml [hunk 1]
```diff
@@ -56,5 +56,7 @@
         <module name="net.bytebuddy"/>
         <module name="java.xml"/>
         <module name="com.h2database.h2" optional="true"/>
+        <module name="com.oracle.ojdbc" optional="true"/>
```
Comment (bstansberry) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/hibernate/main/module.xml:59]: A comment here would be good, e.g.

```
<!-- If the Oracle JDBC driver classes are accessible, Hibernate will use them to tailor OracleDialect behavior -->
```

I assume that statement is accurate; if not we should adapt it. It's deliberately vague to help avoid it falling out of date.

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/hibernate/main/module.xml [hunk 2]
```diff
@@ -56,5 +56,7 @@
...
+        <module name="org.postgresql.jdbc" optional="true"/>
```
Comment (bstansberry) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/hibernate/main/module.xml:60]:
```
<!-- If the Postgres JDBC driver classes are accessible, Hibernate will use them to tailor PostgresDialect and CockroachDialect behavior -->
```

**Review by bstansberry — CHANGES_REQUESTED** [review]

**Comment by bstansberry** [comment]
@jfdenise Please review. I'm mostly interested in your thoughts on whether doing this at all is a good idea, in case you think not!

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/hibernate/main/module.xml (hunk 1 above)
Comment (scottmarlow) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/hibernate/main/module.xml:59]: :+1: for leaving a comment there as that also gives a hint to future maintainers as to why it is there.

Comment (bstansberry) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/hibernate/main/module.xml:60]: Thanks. And yes, maintainability is the main purpose I had in mind. Any optional module dependency begs the question of why it's optional and what conditions would make it required.

**Review by jfdenise — APPROVED** [review]

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @scottmarlow

**Pull request merged**